app/services/custom_landing_page/caching.rb

```diff
@@ -3,23 +3,23 @@ module Caching
 
     module_function
 
-    def fetch_cache_meta(community_id, version, locale)
-      Rails.cache.read("clp/#{community_id}/#{version}/#{locale}")
+    def fetch_cache_meta(community_id, version, locale, cta)
+      Rails.cache.read("clp/#{community_id}/#{version}/#{locale}/#{cta}")
     end
 
-    def fetch_cached_content(community_id, version, digest)
-      Rails.cache.read("clp/#{community_id}/#{version}/#{digest}")
+    def fetch_cached_content(community_id, version, digest, cta)
+      Rails.cache.read("clp/#{community_id}/#{version}/#{digest}/#{cta}")
```

Comment:
I'm not sure you need `cta` in the content cache keys. The content is different, so the digest will be different. As long as you have the meta based on the correct `cta`, it should be fine.